docs/ARCHITECTURE.md

```diff
@@ -1,5 +1,9 @@
 [Back to Guides](README.md)
 
+This document focuses on architecture the 0.10.x version of ActiveModelSerializers. If you are interested in the architecture of the 0.8 or 0.9 versions,
+please refer to the [0.8 README](https://github.com/rails-api/active_model_serializers/blob/0-8-stable/README.md) or
+[0.9 README](https://github.com/rails-api/active_model_serializers/blob/0-9-stable/README.md).
```

Comment:
@groyoh Thanks.  We also want to reference the original design docs which was the impetus for this PR https://github.com/rails-api/active_model_serializers/pull/1490#issuecomment-178665552